### PR TITLE
refactor: extracted code examples

### PR DESCRIPTION
**Pull request by mfranzke**
## Description

As code examples for transformations of input CSS to output CSS are spread through the whole codebase in tests and documentation, we should ensure somehow to reduce redundancies as well as having a single source of truth to correct some code in case of any mistakes or necessary updates.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Performance improvement
- [x] Code refactoring

## Testing

- [x] Tests pass locally
- [ ] Added tests for new functionality
- [ ] Updated existing tests
- [x] Manual testing completed

## Checklist

- [ ] My code follows the project's style guidelines
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published

## Browser Testing

If applicable, please test in the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

## Performance Impact

If this change affects performance:

- [ ] I have run performance benchmarks
- [ ] Performance impact is documented
- [ ] Performance regression is acceptable for the added functionality

## Breaking Changes

If this introduces breaking changes, please describe:

## Additional Notes

Any additional information that reviewers should know.

### CODE REVIEW

File: docs/refactoring/TEST_CENTRALIZATION_SUMMARY.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+1. **More Complex Fixtures**: Add edge cases and error scenarios
+2. **Fixture Validation**: Ensure input/expected pairs are consistent
+3. **Performance Fixtures**: Add timing and benchmark test data
+4. **Interactive Docs**: Web-based fixture browser
+5. **Test Generation**: Auto-generate tests from fixtures
+
+---
+
+**This refactoring establishes a robust, scalable foundation for maintaining CSS test data across the entire css-if-polyfill project while dramatically improving developer experience and documentation quality.**
```
Comment (mfranzke) [inline docs/refactoring/TEST_CENTRALIZATION_SUMMARY.md:261]:
```suggestion
**This refactoring establishes a robust, scalable foundation for maintaining CSS test data across the entire `css-if-polyfill` project while dramatically improving developer experience and documentation quality.**
```

File: docs/refactoring/TEST_CENTRALIZATION_SUMMARY.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+✅ **Cross-Package Reuse**: Fixtures used by both packages
+✅ **Documentation Integration**: Auto-generated from fixtures
+✅ **Maintainability**: Easy to add/modify test cases
+✅ **Zero Regression**: All 103 tests still pass
+✅ **Build Integration**: Seamless CI/CD workflow
+
+## 🔮 **Future Enhancements**
+
+1. **More Complex Fixtures**: Add edge cases and error scenarios
+2. **Fixture Validation**: Ensure input/expected pairs are consistent
```
[the remark this answers is not in the document]
Comment (mfranzke) [reply docs/refactoring/TEST_CENTRALIZATION_SUMMARY.md:255]: https://github.com/mfranzke/css-if-polyfill/issues/38
